ambry-api/src/main/java/com.github.ambry/network/NetworkRequestSend.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+/**
+ * An interface to update send time for a network request
+ */
+public interface NetworkRequestSend {
```

Comment:
If you take a look at how NetworkSend is created at the server side, (line 476 in SocketServer)[https://github.com/linkedin/ambry/blob/master/ambry-network/src/main/java/com.github.ambry.network/SocketServer.java], we don't have an handle to every metric in particular. Its abstracted in a class which exposes method called updateSendTime(value). It might take some effort to fix this. But I don't want to touch the server code for now (since its been running smoothly until now). So, modeled the client also to implement the same api. May be we can sync up offline if you have more doubts